gettsim/policy_environment.py

```diff
@@ -358,9 +361,52 @@ def _load_parameter_group_from_yaml(date, group, parameters=None):
 
     tax_data["datum"] = date
 
+    # Load rounding parameters if they exist
+    if "rounding" in raw_group_data:
+        tax_data["rounding"] = _load_rounding_parameters(
+            date, raw_group_data["rounding"]
+        )
     return tax_data
 
 
+def _load_rounding_parameters(date, rounding_spec):
+    """Load rounding parameters for a specific date from a dictionary.
+
+    Parameters
+    ----------
+    date : datetime.date
+        The date for which the policy system is set up.
+    rounding_spec : dictionary
+        The dictionary that contains rounding parameters
+        for several functions and several dates
+
+    Returns:
+        dictionary: Rounding parameters for the specified date
+    """
+    out = {}
+    rounding_parameters = ["direction", "base"]
+
+    # Load values of all parameters at the specified date
+    for function_name in rounding_spec.keys():
+        dates = sorted(
+            key
+            for key in rounding_spec[function_name].keys()
+            if isinstance(key, datetime.date)
+        )
+        past_policies = [x for x in dates if x <= date]
```

Comment:
I find the logic (too) hard to follow here. Why not get `policy_in_place` in one step? Also, the name `past_policies` suggests that none of them is relevant anymore, maybe that's what is confusing me.

Reply:
Good point about the naming! I improved naming of variables and Documentation.

> Why not get policy_in_place in one step?

To handle cases in which `policy_dates_before_date` is empty. Then taking the maximum leads to an error. (Could be also handled with try except, but I like the current version a bit more.

Alternative would be to fail already here if this happens, but I wouldn't since the user can still change `policy_params` themselves.